fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -286,6 +294,9 @@ public void commitNow() {
                     final long stamp = lock.writeLock();
 
                     try {
+                        if (preCommit != null) {
+                            preCommit.accept(currentContext);
+                        }
                         currentContext.commit();
                         currentContext.close();
```

Comment:
We don't close the context if the `preCommit` fails (or if the commit fails), we should ensure that we always call `close` by putting this in the finally.